.github/workflows/unit.yaml

```diff
@@ -1,37 +1,27 @@
 name: Unit Tests
 
-on:
-  workflow_call:
-    inputs:
-      ref:
-        type: string
-      skipRefCheck:
-        type: boolean
-        default: true
+on: ["pull_request"]
 
 jobs:
   unit-test:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v3
-      with:
-        ref: ${{ inputs.ref }}
 
     - name: Set up Go
-      uses: actions/setup-go@v2
+      uses: actions/setup-go@v3
       with:
         go-version: ~1.20.0
 
     - name: Run Unit Tests
-      if:
-        (github.event_name == 'repository_dispatch' &&
-        github.event.client_payload.slash_command.args.named.sha != '' &&
-        contains(github.event.client_payload.pull_request.head.sha, github.event.client_payload.slash_command.args.named.sha)) ||
-        inputs.skipRefCheck
-      run: go test -race -v ./...
+      run: go test -race -v ./... -coverprofile=coverage.out
+
+    - name: Convert coverage to lcov
+      uses: jandelgado/gcov2lcov-action@v1
 
-    - name: Ensure ref
-      uses: actions/github-script@v6
-      if: ${{ !((github.event_name == 'repository_dispatch' && github.event.client_payload.slash_command.args.named.sha != '' && contains(github.event.client_payload.pull_request.head.sha, github.event.client_payload.slash_command.args.named.sha)) || inputs.skipRefCheck) }}
+    - name: Coveralls
+      uses: coverallsapp/github-action@v2
```

Comment:
can we refer to this action by sha? More secure which is good for all 3rd party actions

Reply:
done, updated. thanks.